src/data/list/rotate.lean

```diff
@@ -376,6 +395,120 @@ begin
   exact ⟨λ ⟨n, hn, h⟩, ⟨n, nat.lt_succ_of_le hn, h⟩, λ ⟨n, hn, h⟩, ⟨n, nat.le_of_lt_succ hn, h⟩⟩
 end
 
+/-- List of all cyclic permutations of `l`.
+The `cyclic_permutations` of a nonempty list `l` will always contain `length l` elements.
+This implies that under certain conditions, there are duplicates in `cyclic_permutations l`.
+The `n`th entry is equal to `l.rotate (n + 1)`, proven in `nth_le_cyclic_permutations`.
+The proof that every cyclic permutant of `l` is in the list is `mem_cyclic_permutations_iff`.
+
+     cyclic_permutations [1, 2, 3, 2, 4] =
+       [[1, 2, 3, 2, 4], [2, 3, 2, 4, 1], [3, 2, 4, 1, 2],
```

Comment:
Can you prove that `head (cyclic_permutations l) = l`? I get the feeling it's not true, meaning this docstring is wrong!

Reply:
You're right. It's the last one. I'll fix the docstring.

Reply:
That is, the proof that the docstring is wrong is `nth_le_cyclic_permutations`

Reply:
Is there a reason not to prefer the definition that gives `[l, l.rotate 1, l.rotate 2, ...]`?

Reply:
I thought the `tail` version would be faster. But I checked with the profiler and it seems the `init` version is about the same speed. So I switched to the simpler one.